src/devices/src/virtio/mmio.rs

```diff
@@ -19,12 +19,72 @@ use crate::bus::BusDevice;
 //TODO crosvm uses 0 here, but IIRC virtio specified some other vendor id that should be used
 const VENDOR_ID: u32 = 0;
 
+/// Interrupt flags (re: interrupt status & acknowledge registers).
+/// See linux/virtio_mmio.h.
+pub const VIRTIO_MMIO_INT_VRING: u32 = 0x01;
+pub const VIRTIO_MMIO_INT_CONFIG: u32 = 0x02;
+
 //required by the virtio mmio device register layout at offset 0 from base
 const MMIO_MAGIC_VALUE: u32 = 0x7472_6976;
 
 //current version specified by the mmio standard (legacy devices used 1 here)
 const MMIO_VERSION: u32 = 2;
 
+/// IrqTrigger specific to the MMIO transport layer
+///
+/// We introduce this struct as an incremental improvement over the old approach where each device
+/// was implementing a `signal_used_queue()` method. Now we can define this logic in one place
+/// in order to avoid code duplication and also we can trigger a guest irq without borrowing
+/// the entire device.
+///
+/// TODO: The irq triggering logic is specific to the transport layer. It would be more natural to
```

Comment:
Can we use a different name ? My view is that the irq triggering is not specific to the transport layer. I am +1 on deduplicating code, but I am not sure we should mix interrupts and transport.

Reply:
Done.